lib/bedrock/data_plane/storage/olivine/database.ex

```diff
@@ -0,0 +1,200 @@
+defmodule Bedrock.DataPlane.Storage.Olivine.Database do
+  @moduledoc false
+
+  @type page_id :: non_neg_integer()
+
+  @opaque t :: %__MODULE__{
+            dets_storage: :dets.tab_name(),
+            window_size_in_microseconds: pos_integer()
+          }
+  defstruct dets_storage: nil,
+            window_size_in_microseconds: 5_000_000
```

Comment:
Any future interest in external configurability of the window size? It's a common request from FDB users, but to be honest the 5s constraint does have positive effects on encouraging good design of data models, IMO. So I'm of two minds about it.